vllm/model_executor/model_loader/weight_utils.py

```diff
@@ -658,8 +658,21 @@ def initialize_dummy_weights(
     for param in model.state_dict().values():
         if torch.is_floating_point(param):
             if current_platform.is_tpu():
-                # XLA device does not support torch.Generator()
-                param.uniform_(low, high)
+                generator = torch.Generator(device="cpu")
+                generator.manual_seed(seed)
+                # Note: The param.uniform_ function cannot be used in this
```

Comment:
how did you find out param.uniform_ will demand more memory?

Reply:
I checked the peak_bytes_used after model loading with the new impl, it's similar to the size of model weights. But for the old impl, it's OOM.

Reply:
but can you explain a bit why `param.copy_` fixes the issue?

Reply:
`uniform_` might have complex computation logic in vLLM, `copy_` is basically a CPU to TPU transfer and update the version of XLA tensor